### PR TITLE
Added support for XPath selectors

### PR DESCRIPTION
**Pull request by pjforde1978**
I am excited to propose the addition of support for XPath selectors in addition to CSS selectors to the cable_ready API. This will facilitate some exciting features on related projects such as stimulus_reflex.

This change is non-breaking, as it still expects a CSS selector by default.

However, if you happen to pass in an XPath query that resolves to a single DOM element via config.selector, you can set the new config.xpath boolean option to true and it will handle it no problem.

### CODE REVIEW

**Comment by hopsoft** [comment]
This is great. I'm happy to accept but ask that you first merge master and then run `bin/standardize`. Also that you add a section about using XPath to the readme. Thanks.

**Comment by hopsoft** [comment]
🤦‍♂ Looks like I created a conflict for you. Sorry about that.

File: bin/standardize [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
 bundle exec standardrb --fix
-cd ./javascript && yarn run prettier --write cable_ready.js
+cd ./javascript && yarn run prettier --write cable_ready.js
```
Comment (pjforde1978) [inline bin/standardize]: I don't understand how this is different. I don't think the file had a newline before?

**Comment by pjforde1978** [comment]
Even though Code Climate disagrees, I believe that my changes to the JS library and the README are ready for review. StandardRB did raise some warnings but they were idiomatic and would require breaking your API.

**Pull request merged**